internal/graph/check.go

```diff
@@ -768,8 +768,19 @@ func (c *LocalChecker) checkDirect(parentctx context.Context, req *ResolveCheckR
 		resp, err := union(ctx, c.concurrencyLimit, checkFuncs...)
 		if err != nil {
 			telemetry.TraceError(span, err)
+			return nil, err
 		}
-		return resp, err
+
+		// count db reads after they happen in the case that we didn't find 'allowed=false' but we still incurred reads
+		if len(directlyRelatedUsersetTypes) > 0 {
+			// if we had N userset checks, that was 1 read, not N
+			resp.GetResolutionMetadata().DatastoreQueryCount++
+		}
+		if shouldCheckDirectTuple {
+			resp.GetResolutionMetadata().DatastoreQueryCount++
```

Comment:
Similar to the comment above, this should be part of responsibility of fn1 to increment the datastore query count

Reply:
undid this change